packages/compiler-cli/src/ngtsc/annotations/src/directive.ts

```diff
@@ -437,7 +437,15 @@ function extractHostBindings(
     });
   }
 
-  const {attributes, listeners, properties} = parseHostBindings(hostMetadata);
+  const bindings = parseHostBindings(hostMetadata);
+
+  // Create and provide proper sourceSpan to make error message more descriptive (FW-995)
+  const errors = verifyHostBindings(bindings, null !);
+  if (errors.length) {
```

Comment:
Use a real check instead of shortcut code:

```suggestion
  if (errors.length > 0) {
```